src/sudoers/mod.rs

```diff
@@ -208,28 +208,19 @@ fn group_cmd_specs_per_runas<'a>(
     cmnd_specs: impl Iterator<Item = (Option<&'a RunAs>, (Tag, &'a Spec<Command>))>,
     cmnd_aliases: &'a [Def<Command>],
 ) -> impl Iterator<Item = Entry<'a>> {
+    use std::borrow::BorrowMut;
+    let mut cmnd_specs = cmnd_specs.peekable();
     let mut entries = vec![];
-    let mut last_runas = None;
-    let mut collected_specs = vec![];
-
-    for (runas, (tag, spec)) in cmnd_specs {
-        if runas.map(|r| r as *const _) != last_runas.map(|r| r as *const _) {
-            if !collected_specs.is_empty() {
-                entries.push(Entry::new(
-                    last_runas,
-                    mem::take(&mut collected_specs),
-                    cmnd_aliases,
-                ));
-            }
-
-            last_runas = runas;
-        }
-
-        collected_specs.push((tag, spec));
-    }
 
-    if !collected_specs.is_empty() {
-        entries.push(Entry::new(last_runas, collected_specs, cmnd_aliases));
+    while let Some(&(cur_runas, _)) = cmnd_specs.peek() {
+        let specs = cmnd_specs.borrow_mut().take_while(|(runas, _)| {
+            runas.map(|r| r as *const _) == cur_runas.map(|r| r as *const _)
+        });
```

Comment:
`take_while` will still consume an element from the underlying iterator once the condition returns false, so the current code will silently drop an element whenever you go to a different group.

Reply:
Yep. Any suggestion on changing that? I prefer the more declarative style of the routine but I had hoped that `.take_while()` would respect the `Peekable` semantics. But alas.

Reply:
You could use a loop in which you peek manually and only `.next()` when you want to consume the element. I don't think there is a way to write this with iterator chains though without using an external crate like itertools or without writing our own iterator combinator for this, neither of which are solutions I would prefer.

Reply:
Yes, writing my own `chunk_by` is fun and all, but not something I wanted to do here. In that case I'm letting this one go.